sycl/source/detail/scheduler/graph_builder.cpp

```diff
@@ -269,220 +271,114 @@ Command *Scheduler::GraphBuilder::addCopyBack(Requirement *Req) {
 // destruction the memory should be written back(if access mode is not read
 // only) to the memory object. No operations with buffer allowed during host
 // accessor lifetime.
-Command *Scheduler::GraphBuilder::addHostAccessor(Requirement *Req,
-                                                  EventImplPtr &RetEvent) {
-  QueueImplPtr HostQueue = Scheduler::getInstance().getDefaultHostQueue();
+Command *Scheduler::GraphBuilder::addHostAccessor(Requirement *Req) {
+  const QueueImplPtr &HostQueue =
+      Scheduler::getInstance().getDefaultHostQueue();
   MemObjRecord *Record = getOrInsertMemObjRecord(HostQueue, Req);
+
   if (MPrintOptionsArray[BeforeAddHostAcc])
     printGraphAsDot("before_addHostAccessor");
   markModifiedIfWrite(Record, Req);
-  QueueImplPtr SrcQueue;
 
-  std::set<Command *> Deps = findDepsForReq(Record, Req, HostQueue);
+  AllocaCommandBase *SrcAllocaCmd = nullptr;
   // If we got freshly new Record, there is no point in searching
   // for allocas on device.
   if (Record->MAllocaCommands.empty())
-    SrcQueue = HostQueue;
+    SrcAllocaCmd = getOrCreateAllocaForReq(Record, Req, HostQueue);
   else
-    SrcQueue = (*Deps.begin())->getQueue();
+    SrcAllocaCmd = findAllocaForReq(Record, Req, Record->MCurContext);
```

Comment:
Is it possible to replace these lines with only line 287?

Reply:
It should be. Something started to crash. Will try again.

Reply:
No, it's not possible as currently getOrCreateAlloca takes queue while findAllocaForReq takes context.